types/lodash/common/object.d.ts

```diff
@@ -1987,32 +1987,47 @@ declare module "../index" {
 
     // invert
 
+    // helpers
+    type InvertAllValues<T extends Record<PropertyKey, PropertyKey>> = {
+        [P in keyof T]: { key: P, value: T[P] }
+    }[keyof T]
+    type InvertResult<T extends Record<PropertyKey, PropertyKey>> = {} & {
+        [P in InvertAllValues<T>['value']]: Extract<InvertAllValues<T>, { value: P }>['key']
+    }
```

Comment:
Tests are running out of memory so we need to find a simpler way of doing this. Would something like this work?

```ts
type Invert<T> = T extends Record<infer U, infer V> ? Record<V, U> : Dictionary<string>;
```

I'm not sure if TS will like that though. Also, even if it compiles, it will lose some information - your test ocject would probably get mapped to something like `{ x: "a" | "b", y: "a" | "b" }`, which is an improvement over the old version, but not perfect.